### DISCUSSION TITLE
Slow read write with Prisma

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#1638] [Questions] [answered]
Hello all, 

New member here. New to Supabase as well. I just recently spun off a database for development, what I immediately noticed was the very slow read write speeds. I see at least a gap of 2 seconds for both read and write. My intention is to develop a chat app. If this is the way it is going to work in production, I might have to move off Supabase. Could somebody please confirm, this wont be the case?

Thanks.

**Comment by steve-chavez** [comment, marked as answer]
> what I immediately noticed was the very slow read write speeds. I see at least a gap of 2 seconds for both read and write

How are you making the reads and writes?

There are performance numbers here: https://supabase.io/beta#performance

**Reply by sumchans** [reply]
Running queries through Prisma, connected it directly to the Supabase Postgres database.

**Reply by steve-chavez** [reply]
Maybe you're opening and closing connections on each query? Maybe post an example of how are you making those queries.

Also, connecting through Supabase pgbouncer is recommended for Prisma. See https://supabase.io/blog/2021/04/02/supabase-pgbouncer

**Reply by sumchans** [reply]
I have checked the pool method, it was already set to transaction, I believe is the recommended option. when doing queries from Prisma.

This is code I wrote to create a record - 
```
```
const prisma = new PrismaClient()

async function main() {
  // ... you will write your Prisma Client queries here
  await prisma.user.create({
    data: {
      name: 'Sumesh',
      email: 'sumesh@test.com',
      posts: {
        create: { title: 'Hello World' },
      },
      profile: {
        create: { bio: 'I like turtles' },
      },
    },
  })

  const allUsers = await prisma.user.findMany({
    include: {
      posts: true,
      profile: true,
    },
  })
  console.dir(allUsers, { depth: null })
}

main()
  .catch((e) => {
    throw e
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

```

**Reply by steve-chavez** [reply]
>  I see at least a gap of 2 seconds for both read and write.

How do you measure that time?

**Reply by sumchans** [reply]
When I do a query, I see the results are returned with a lag of 2 seconds.

**Reply by steve-chavez** [reply]
>  I see the results are returned with a lag of 2 seconds.

Some code samples would help, that way I can see what's wrong.

However, at worst, latency should be in milliseconds. Check this previous discussion where a user was using a Pandas client: https://github.com/supabase/supabase/discussions/577#discussioncomment-302817
